src/mobu/models/business/notebookrunner.py

```diff
@@ -43,12 +43,27 @@ class NotebookRunnerOptions(NubladoBusinessOptions):
         description="Only used by the NotebookRunner",
     )
 
+    repo_ref: str | None = Field(
+        None,
+        title="Git ref of notebook repository to execute",
+        description="Only used by the NotebookRunner",
+    )
```

Comment:
You probably need a validator here to reject a configuration that sets both branch and ref, since I think that's nonsensical. Or maybe just drop the branch case entirely and only take refs, and always use the clone and then checkout pattern.